### PR TITLE
chore: Added tests for GetAdminUrlForLanguage template tag

### PR DESCRIPTION
**Pull request by filipweidemann**
## Description
This PR catches up on #8044 and adds two new tests targeting the regressed template tag `GetAdminUrlForLanguage`.
I think this is worth it. Please let me know what you think.

## Related resources
#8044 

## Checklist
* [x] I have opened this pull request against ``develop-4``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined the channel #pr-reviews on our [Discord Server](https://discord-pr-review-channel.django-cms.org) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: cms/tests/test_templatetags.py [hunk 1]
```diff
@@ -43,6 +43,48 @@
...
+        # German PageContent will have pk=2
+        german_content = create_page_content("de", "AdminURLTestPage German Content", page)
+
+
+        request = RequestFactory().get('/')
+        request.current_page = page
+        with force_language('de'):
+            url = GetAdminUrlForLanguage.get_value(None, context={'request': request}, page=page, language="de")
+            # We expect that getting the German URL will return the German page content (change link, not a create link)
+            self.assertEqual(url, '/de/admin/cms/pagecontent/2/change/')
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:59]:
```suggestion
            self.assertEqual(url, f'/de/admin/cms/pagecontent/{german_content.pk}/change/')
```
It's unsafe to make assumptions on pk's in testing environments.

File: cms/tests/test_templatetags.py [hunk 2]
```diff
@@ -43,6 +43,48 @@
...
+
+        with force_language('en'):
+            url = GetAdminUrlForLanguage.get_value(None, context={'request': request}, page=page, language="en")
+            # Same goes for the English URL
+            self.assertEqual(url, '/en/admin/cms/pagecontent/1/change/')
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:64]:
```suggestion
            self.assertEqual(url, f'/en/admin/cms/pagecontent/{english_content.pk}/change/')
```

File: cms/tests/test_templatetags.py [hunk 3]
```diff
@@ -43,6 +43,48 @@
 
 
 class TemplatetagTests(CMSTestCase):
+    def test_get_admin_url_for_language(self):
+        # This creates a new page and the corresponding PageContent object for lang 'en', both with pk=1
+        page = create_page("AdminURLTestPage English Content", "nav_playground.html", "en")
+
+        # German PageContent will have pk=2
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:50]:
```suggestion
```
It will not necessarily.

File: cms/tests/test_templatetags.py [hunk 4]
```diff
@@ -43,6 +43,48 @@
 
 
 class TemplatetagTests(CMSTestCase):
+    def test_get_admin_url_for_language(self):
+        # This creates a new page and the corresponding PageContent object for lang 'en', both with pk=1
+        page = create_page("AdminURLTestPage English Content", "nav_playground.html", "en")
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:48]:
```suggestion
        page = create_page("AdminURLTestPage English Content", "nav_playground.html", "en")
        
        english_content = page.pagecontent_set(manager="admin_manager").first()
```

File: cms/tests/test_templatetags.py [hunk 5]
```diff
@@ -43,6 +43,48 @@
...
+            self.assertEqual(url, '/en/admin/cms/pagecontent/1/change/')
+
+        with force_language('fr'):
+            url = GetAdminUrlForLanguage.get_value(None, context={'request': request}, page=page, language="fr")
+            # PageContent for the French languag does not exist, we expect a link to create a new PageContent instance
+            self.assertEqual(url, '/fr/admin/cms/pagecontent/add/?cms_page=1&language=fr')
+
+    def test_admin_pagecontent_language_tab_urls(self):
+        # Same setup as in the templatetag test above
+        page = create_page('Test', 'nav_playground.html', 'en')
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:73]:
```suggestion
        page = create_page('Test', 'nav_playground.html', 'en')
        english_coontent = page.pagecontent_set(manager="admin_manager").first()
```

File: cms/tests/test_templatetags.py [hunk 6]
```diff
@@ -43,6 +43,48 @@
...
+        request = RequestFactory().get('/')
+        request.current_page = page
+        template = """
+            {% load cms_tags cms_admin %}
+            {% get_admin_url_for_language page_obj 'en' %}
+            {% get_admin_url_for_language page_obj 'de' %}
+            {% get_admin_url_for_language page_obj 'fr' %}
+        """
+        output = self.render_template_obj(template, {'page_obj': page}, request)
+        self.assertIn('/en/admin/cms/pagecontent/1/change/', output)
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:85]:
```suggestion
        self.assertIn(f'/en/admin/cms/pagecontent/{english_content.pk}/change/', output)
```

File: cms/tests/test_templatetags.py [hunk 7]
```diff
@@ -43,6 +43,48 @@
...
+        self.assertIn('/en/admin/cms/pagecontent/2/change/', output)
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:86]:
```suggestion
        self.assertIn(f'/en/admin/cms/pagecontent/{german_content.pk}/change/', output)
```

**Review by fsbraun — COMMENTED** [review]
Yes, this is a good idea! 

However, you'll need to avoid assumptions on the primary keys.

**Comment by filipweidemann** [comment]
@fsbraun Yup, that's right. I added your suggestions in one commit & fixed a spelling issue. Should be fine now! Do you have any other suggestions?

**Comment by fsbraun** [comment]
@filipweidemann What exactly is the difference between the two tests? Do we need them both?

I can imagine two tests (based on either of yours).
1. In the test setup call `page.get_content_obj("en")` before calling the template tag.
2. In the test setup call `page.get_admin_content("en")` before calling the template tag.

With a partially filled cache by the two calls, the tests would fail.

**Comment by filipweidemann** [comment]
@fsbraun Yeah wow, I actually forgot the third case, you're right. Was thinking about this before implementing it. I'll add it as well!

About the current ones: I can probably remove the first one and only stick with the "faked" template one. My idea was that we also check if it can be forced to craft the links according to our language but let's be real, it actually is not doing anything here because it uses Django's `reverse` function. So what I am actually testing is Django and yeah, this is kind of unnecessary.

I'll remove the first one and replace it with the one you suggested because this one is actually missing right now!

**Comment by filipweidemann** [comment]
@fsbraun okay so I deleted the first one and added the method calls before rendering the template.

I also verified that this catches the regression by temporarily adding `del self.admin_content_cache["de"]` inside the `set_admin_content_cache`. The test immediately failed.

This is cool. Anything else?

File: cms/tests/test_templatetags.py [hunk 8]
```diff
@@ -43,6 +43,28 @@
...
 
 class TemplatetagTests(CMSTestCase):
+    def test_admin_pagecontent_language_tab_urls(self):
+        # Same setup as in the templatetag test above
+        page = create_page('AdminURLTestPage English Content', 'nav_playground.html', 'en')
+        english_content = page.pagecontent_set(manager="admin_manager").first()
+        german_content = create_page_content("de", "AdminURLTestPage German Content", page)
+
+        # Try to fill the cache with partial data (this should not be possible)
+        page.get_content_obj(language='en')
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:53]:
```suggestion
        page.get_content_obj(language='en')  # should not affect admin template tag
```

File: cms/tests/test_templatetags.py [hunk 9]
```diff
@@ -43,6 +43,28 @@
...
+        page.get_admin_content(language='en')
```
Comment (fsbraun) [inline cms/tests/test_templatetags.py:54]:
```suggestion
        page.get_admin_content(language='en')  # Should fill the whole cache
```

**Review by fsbraun — APPROVED** [review]
Nice! Looks good to me!! Thank you so much!

**Pull request merged**